src/LiveComponent/src/LiveCollectionTrait.php

```diff
@@ -27,30 +27,39 @@ trait LiveCollectionTrait
     #[LiveAction]
     public function addCollectionItem(PropertyAccessorInterface $propertyAccessor, #[LiveArg] string $name): void
     {
-        if (str_starts_with($name, $this->formName)) {
-            $name = substr_replace($name, '', 0, mb_strlen($this->formName));
-        }
-
-        $data = $propertyAccessor->getValue($this->formValues, $name);
+        $propertyPath = $this->fieldNameToPropertyPath($name, $this->formName);
+        $data = $propertyAccessor->getValue($this->formValues, $propertyPath);
 
         if (!\is_array($data)) {
-            $propertyAccessor->setValue($this->formValues, $name, []);
+            $propertyAccessor->setValue($this->formValues, $propertyPath, []);
             $data = [];
         }
 
         $index = [] !== $data ? max(array_keys($data)) + 1 : 0;
-        $propertyAccessor->setValue($this->formValues, $name."[$index]", []);
+        $propertyAccessor->setValue($this->formValues, $propertyPath."[$index]", []);
     }
 
     #[LiveAction]
     public function removeCollectionItem(PropertyAccessorInterface $propertyAccessor, #[LiveArg] string $name, #[LiveArg] int $index): void
     {
-        if (str_starts_with($name, $this->formName)) {
-            $name = substr_replace($name, '', 0, mb_strlen($this->formName));
+        $propertyPath = $this->fieldNameToPropertyPath($name, $this->formName);
+        $data = $propertyAccessor->getValue($this->formValues, $propertyPath);
+        unset($data[$index]);
```

Comment:
What if array_splice would be used to reindex?

Reply:
What would be the benefit of that? I think it's more readable this way. We are just removing an item here, not re-indexing the array.

Reply:
Most of the time you'd want a list, no?

Reply:
If we re-index the array here that would trigger re-rendering all other items after the removed item on the client, AFAIK. I had no issues with this so far, so I rather not do that.

Reply:
Hm,okay,will see if that works afterwards